lib/rules/no-array-index-key.js

```diff
@@ -16,9 +16,10 @@ const pragma = require('../util/pragma');
 
 module.exports = {
   meta: {
+    type: 'problem',
     docs: {
       description: 'Prevent usage of Array index in keys',
-      category: 'Best Practices',
+      category: 'Possible Errors',
```

Comment:
```suggestion
      category: 'Best Practices',
```
it's not an error to use array indexes, it just defeats the point of `key`.

Reply:
This may result in items not being rendered.(https://reactjs.org/docs/reconciliation.html#recursing-on-children)

Reply:
> This can work well if the items are never reordered, but reorders will be slow.

Can you provide an example of how it could cause that problem?

Reply:
Here: https://jsbin.com/wohima/edit?output

Reply:
Here: https://jsbin.com/wohima/edit?output

Reply:
Thanks, that's pretty clear (altho adding an item to the beginning of the list seems like terrible UX).